qiskit_ibm_provider/jupyter/live_data_widget.py

```diff
@@ -0,0 +1,1385 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A module for visualizing device coupling maps"""
+
+from enum import Enum
+from datetime import datetime
+import asyncio
+import json
+import ssl
+import zlib
+from io import BytesIO
+from base64 import b64encode
+import logging
+import pytz
+from websocket import WebSocketApp
+import ipywidgets as widgets
+import numpy as np
+from sklearn.decomposition import PCA
```

Comment:
Why is `sklearn` declared as a development dependency but used in production code?

Reply:
I'm not sure why this was done originally in `ibmq-provider` - should it be moved to `requirements.txt`?

Reply:
If we're not sure if it's needed as a production dependency I'd prefer to keep it as development dependency until proven otherwise.